src/dashboard/user-interface/scores/scores-route-trait.php

```diff
@@ -0,0 +1,234 @@
+use Yoast\WP\SEO\Dashboard\Infrastructure\Content_Types\Content_Types_Collector;
+use Yoast\WP\SEO\Main;
+use Yoast\WP\SEO\Repositories\Indexable_Repository;
+
+/**
+ * Trait for routes of scores.
+ */
+trait Scores_Route_Trait {
+
+	/**
+	 * The content types collector.
+	 *
+	 * @var Content_Types_Collector
+	 */
+	protected $content_types_collector;
+
+	/**
+	 * The taxonomies repository.
+	 *
+	 * @var Taxonomies_Repository
+	 */
+	protected $taxonomies_repository;
+
+	/**
+	 * The indexable repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	protected $indexable_repository;
+
+	/**
+	 * The scores repository.
+	 *
+	 * @var Scores_Repository_Interface
+	 */
+	private $scores_repository;
+
+	/**
+	 * Sets the collectors for the trait.
+	 *
+	 * @required
+	 *
+	 * @param Content_Types_Collector $content_types_collector The content type collector.
+	 *
+	 * @return void
+	 */
+	public function set_collectors(
+		Content_Types_Collector $content_types_collector
+	) {
+		$this->content_types_collector = $content_types_collector;
+	}
+
+	/**
+	 * Sets the collectors for the trait.
+	 *
+	 * @required
+	 *
+	 * @param Taxonomies_Repository $taxonomies_repository The taxonomies repository.
+	 * @param Indexable_Repository  $indexable_repository  The indexable repository.
+	 *
+	 * @return void
+	 */
+	public function set_repositories(
+		Taxonomies_Repository $taxonomies_repository,
+		Indexable_Repository $indexable_repository
+	) {
+		$this->taxonomies_repository = $taxonomies_repository;
+		$this->indexable_repository  = $indexable_repository;
+	}
+
+	/**
+	 * Registers routes for scores.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			Main::API_V1_NAMESPACE,
+			self::ROUTE_PREFIX,
+			[
+				[
+					'methods'             => 'GET',
+					'callback'            => [ $this, 'get_scores' ],
+					'permission_callback' => [ $this, 'permission_manage_options' ],
+					'args'                => [
+						'contentType' => [
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+						],
+						'taxonomy' => [
+							'required'          => false,
+							'type'              => 'string',
+							'default'           => '',
+							'sanitize_callback' => 'sanitize_text_field',
+						],
+						'term' => [
+							'required'          => false,
+							'type'              => 'integer',
+							'default'           => null,
+							'sanitize_callback' => static function ( $param ) {
+								return \intval( $param );
+							},
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Gets the scores of a specific content type.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The success or failure response.
+	 */
+	public function get_scores( WP_REST_Request $request ) {
+		$content_type = $this->get_content_type( $request['contentType'] );
```

Comment:
You could choose to work with exceptions here instead of doing good 'ol if checks to determine your response. The exception can have the error message and proper status code and then you can just do a try catch here and die the if checks in the responsible get* methods

Reply:
Yeah, [done](https://github.com/Yoast/wordpress-seo/pull/21847/commits/19c8839603efc560f2c4876986c058a1a6e9fe8b)